### PR TITLE
Refactored to map_join; consolidated redundancy

### PR DESCRIPTION
**Pull request by bradhanks**
I made a couple changes suggested by credo and also changed two functions from Enum to Stream because it ends with Enum.uniq

### CODE REVIEW

File: lib/hexpm/block_address/block_address.ex [hunk 1]
```diff
@@ -20,8 +20,8 @@ defmodule Hexpm.BlockAddress do
     disallowed =
       Hexpm.BlockAddress.Entry
       |> Hexpm.Repo.all()
-      |> Enum.map(&Hexpm.Utils.parse_ip_mask(&1.ip))
-      |> Enum.reject(fn {ip, _mask} -> ip == nil end)
+      |> Stream.map(&Hexpm.Utils.parse_ip_mask(&1.ip))
+      |> Stream.reject(fn {ip, _mask} -> ip == nil end)
```
Comment (wojtekmach) [inline lib/hexpm/block_address/block_address.ex:24]: A for/1 with uniq: true would be even better

File: lib/hexpm_web/views/email_view.ex [hunk 2]
```diff
@@ -66,8 +66,7 @@ defmodule HexpmWeb.EmailView do
 
     def table(candidates) do
       candidates
-      |> Enum.map(fn [n, c, d] -> "#{n},#{c},#{d}" end)
-      |> Enum.join("\n")
+      |> Enum.map_join("\n", fn [n, c, d] -> "#{n},#{c},#{d}" end)
```
Comment (wojtekmach) [inline lib/hexpm_web/views/email_view.ex:69]: Let’s avoid single pipelines

File: lib/hexpm/block_address/block_address.ex (hunk 1 above)
Comment (bradhanks) [reply lib/hexpm/block_address/block_address.ex]: ok I made those two changes let me know what you think.

File: lib/hexpm/block_address/block_address.ex [hunk 3]
```diff
@@ -18,11 +18,12 @@ defmodule Hexpm.BlockAddress do
 
   def reload() do
     disallowed =
-      Hexpm.BlockAddress.Entry
-      |> Hexpm.Repo.all()
-      |> Enum.map(&Hexpm.Utils.parse_ip_mask(&1.ip))
-      |> Enum.reject(fn {ip, _mask} -> ip == nil end)
-      |> Enum.uniq()
+      for entry <- Hexpm.BlockAddress.Entry |> Hexpm.Repo.all(),
```
Comment (wojtekmach) [inline lib/hexpm/block_address/block_address.ex:21]:
```suggestion
      for entry <- Hexpm.Repo.all(Hexpm.BlockAddress.Entry),
```

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**